app/main.py

```diff
@@ -1 +1,23 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import (
+    NotVaccinatedError,
+    OutdatedVaccineError,
+    NotWearingMaskError
+)
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+    masks_needed = 0
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except NotVaccinatedError as e:
+            return e.__str__()
```

Comment:
Do not call dunder methods when it is possible to avoid this. Call `str()` on errors or define the string that should be returned out of the scope of the error implementation.